lucille-core/src/main/java/com/kmwllc/lucille/connector/AbstractConnector.java

```diff
@@ -32,13 +32,21 @@ public abstract class AbstractConnector implements Connector {
   private String message = null;
   protected final Config config;
 
-  public AbstractConnector(Config config, Spec spec) {
+  public AbstractConnector(Config config) {
     this.config = config;
     this.name = config.getString("name");
     this.pipelineName = config.hasPath("pipeline") ? config.getString("pipeline") : null;
     this.docIdPrefix = config.hasPath("docIdPrefix") ? config.getString("docIdPrefix") : "";
     this.collapse = config.hasPath("collapse") ? config.getBoolean("collapse") : false;
 
+    Spec spec;
+
+    try {
+      spec = getSpec();
+    } catch (Exception e) {
+      throw new RuntimeException("Error accessing " + getClass() + " Spec. Is it publicly and statically available under \"SPEC\"?", e);
```

Comment:
Suggest moving this try/catch into getSpec() 
could remove "throws Exception" from getSpec() since we'd be throwing an unchecked RuntimeException